Cython/Compiler/ModuleNode.py

```diff
@@ -1875,20 +1873,19 @@ def generate_clear_function(self, scope, code, cclass_entry):
         have_entries, (py_attrs, py_buffers, memoryview_slices) = (
             scope.get_refcounted_entries(include_gc_simple=False))
 
-        if py_attrs or py_buffers or base_type:
-            unused = ''
-        else:
-            unused = 'CYTHON_UNUSED '
-
         code.putln("")
-        code.putln("static int %s(%sPyObject *o) {" % (slot_func, unused))
+        code.putln("static int %s(PyObject *o) {" % slot_func)
 
         if py_attrs and Options.clear_to_none:
             code.putln("PyObject* tmp;")
 
         if py_attrs or py_buffers:
             self.generate_self_cast(scope, code)
 
+        # silence unused warnings about unused variables
+        if not py_attrs and not py_buffers and not base_type:
```

Comment:
I prefer keeping the tests similar to what we use for declaring and using these variables.
```suggestion
        if not (py_attrs or py_buffers or base_type):
```